app/src/theme/index.tsx

```diff
@@ -524,3 +524,86 @@ export function ThemeProvider({ children }: { children: React.ReactNode }) {
 
   return <StyledThemeProvider theme={themeAggregator(darkMode)}>{children}</StyledThemeProvider>
 }
+
+const TextWrapper = styled.div<{ color: keyof Colors }>`
+  color: ${({ color, theme }) => (theme as any)[color]};
+`
+
+export const TYPE = {
+  heading1(props: any) {
+    return (
+      <TextWrapper
+        align={'left'}
+        color={'text3'}
+        fontFamily={'Roboto'}
+        fontSize={'22px'}
+        fontWeight={500}
+        letterSpacing={'0.8px'}
+        lineHeight={'100% || 25.78px'}
+        verticalAlign={'center'}
+        {...props}
+      />
+    )
+  },
+  heading2(props: any) {
+    return (
+      <TextWrapper
+        align={'left'}
+        color={'text3'}
+        fontFamily={'Roboto'}
+        fontSize={'18px'}
+        fontWeight={400}
+        letterSpacing={'0.2px'}
+        lineHeight={'100% || 21.09px'}
+        verticalAlign={'center'}
+        {...props}
+      />
+    )
+  },
+  heading3(props: any) {
+    return (
+      <TextWrapper
+        align={'left'}
+        color={'text3'}
+        fontFamily={'Roboto'}
+        fontSize={'16px'}
+        fontWeight={500}
+        letterSpacing={'0.2px'}
+        lineHeight={'100% || 18.75px'}
+        paragraphSpacing={'4px'}
+        verticalAlign={'top'}
+        {...props}
+      />
+    )
+  },
+  bodyMedium(props: any) {
+    return (
+      <TextWrapper
+        align={'left'}
+        color={'text3'}
+        fontFamily={'Roboto'}
+        fontSize={'14px'}
+        fontWeight={500}
+        letterSpacing={'0.2px'}
+        lineHeight={'100% || 16.41px'}
+        verticalAlign={'top'}
+        {...props}
+      />
+    )
+  },
+  bodyRegular(props: any) {
+    return (
+      <TextWrapper
+        align={'left'}
+        color={'text3'}
+        fontFamily={'Roboto'}
+        fontSize={'14px'}
+        fontWeight={400}
+        letterSpacing={'0.2px'}
+        lineHeight={'100% || 16.41px'}
+        verticalAlign={'top'}
+        {...props}
```

Comment:
@Frankaus All in all this how it should be implemented. But one important thing to keep in mind which is crucial is that TextWrapper should be like base for all the shared styling that all text should include as you see you are repeating couple of properties on all of them like font-family, align,letter-spacing...Always keep DRY(dont repeat yourself) principle in mind one of the most important things in programming...